solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -114,6 +114,51 @@ pub mod audius_data {
         Ok(())
     }
 
+    /// Functionality to create user without admin privileges
+    pub fn create_user(
+        ctx: Context<CreateUser>,
+        base: Pubkey,
+        eth_address: [u8; 20],
+        handle_seed: [u8; 16],
+        _user_bump: u8,
+        metadata: String,
+        user_authority: Pubkey,
+    ) -> ProgramResult {
+        msg!("Audius::CreateUser");
+
+        // Confirm that the derived pda from base is the same as the user storage account
+        let (derived_user_acct, _) = Pubkey::find_program_address(
+            &[&base.to_bytes()[..32], &handle_seed],
+            ctx.program_id,
+        );
+        if derived_user_acct != ctx.accounts.user.key() {
```

Comment:
isn't this derivation verified by the account constraints? what is the appropriate validation here around the base account?

Reply:
@cheran-senthil - does this still make sense given the account constraints you've added here? https://github.com/AudiusProject/audius-protocol/pull/2294/files#diff-74cd8acfd34a7b82ca1efedf797bd58e7321341013706e0f8d74684fec16fe4eR383

Reply:
I removed this validation